File: FRONTEND/app/src/main/java/com/myscratch/app/domain/model/VaultItem.kt
package com.myscratch.app.domain.model

enum class VaultCategory(val key: String, val title: String) {
    ACCOUNT("account", "Akun & Password"),
    BANK("bank", "Kartu & Bank"),
    IDENTITY("identity", "Identitas"),
    NOTE("note", "Catatan Rahasia");

    companion object {
        fun fromKey(key: String): VaultCategory =
            entries.find { it.key.equals(key, ignoreCase = true) } ?: ACCOUNT
    }
}

data class VaultItem(
    val id: String,
    val userId: String,
    val category: String,
    val title: String,
    val username: String? = null,
    val password: String? = null,
    val extraData: String? = null,
    val notes: String? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

File: BACKEND/app/Models/VaultItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class VaultItem extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'user_id',
        'category',
        'title',
        'username',
        'password',
        'extra_data',
        'notes',
    ];

    protected $casts = [
        'title' => 'encrypted',
        'username' => 'encrypted',
        'password' => 'encrypted',
        'extra_data' => 'encrypted',
        'notes' => 'encrypted',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}
